### PR TITLE
daily cmake updates

### PR DESCRIPTION
**Pull request by madebr**
- No need to prepend compile definitions with `/D` or `-D` in cmake
- Link privately to crashpad client (#221)

- tests don't need c++14
- compile options of sentry can private + only include directory should be public

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -284,7 +284,7 @@ if(SENTRY_BACKEND_CRASHPAD)
 			set(CRASHPAD_ENABLE_INSTALL ON CACHE BOOL "Enable crashpad installation" FORCE)
 		endif()
 		add_subdirectory(external/crashpad crashpad_build)
-		target_link_libraries(sentry PUBLIC
+		target_link_libraries(sentry PRIVATE
```
Comment (Swatinem) [inline CMakeLists.txt:287]: When building a shared lib, this makes total sense.
When building a static lib, how can we make sure that users will transitively get crashpad as well? Or does that happen already?

Comment (madebr) [reply CMakeLists.txt:287]: CMake does this automatically.
I checked the generated targets and it contains a `$<LINK_ONLY:...>` in the generated targets file (in the `INTERFACE_LINK_LIBRARIES` property).
This is similar to the following situation:
```
add_library(liba STATIC ...)
add_library(libb STATIC ...)
target_link_libraries(libb PRIVATE liba)
add_executable(main ...)
target_link_libraries(main PRIVATE libb)
```
The `main` executable must link to `liba ` and `libb`.

Comment (madebr) [reply CMakeLists.txt:287]: cmake knows static libraries should be made transparent.

Comment (madebr) [reply CMakeLists.txt:287]: @Swatinem 
I just noticed I introduced a bug in crashpad yesterday.
I used `CMAKE_C_FLAGS` twice as source for the regex.
I'll open a pr...

Comment (madebr) [reply CMakeLists.txt:287]: Ok, I"ll bump crashpad after my fix to crashpad is merged.

**Comment by madebr** [comment]
Question: what c standard are you targeting c99 or c11?

File: CMakeLists.txt (hunk 1 above)
Comment (madebr) [reply CMakeLists.txt:289]: See getsentry/crashpad#9

**Comment by Swatinem** [comment]
> Question: what c standard are you targeting: c99 or c11?

If it were up to me personally, I would go with the most recent one ;-)
I guess since we have c++14, we should be safe to require c11. I think @jan-auer known more about this than I do though.

**Comment by madebr** [comment]
I've updated the crashpad git submodule.

I ask about the c standard so all compilers now and in the future will know what standard to use.
This standard will also be set PRIVATE and not visible to consumers.

**Comment by Swatinem** [comment]
> I've updated the crashpad git submodule.

would you mind bumping again, to also pull in the changes from @Amphaal in one go?

**Comment by Amphaal** [comment]
Btw many thanks to @madebr who basically did what I wanted to do a while ago :') I can now safely delete my forks 👍

**Comment by Swatinem** [comment]
Thanks to both of you! I am very grateful for these contributions.

**Pull request merged**